src/System.Net.Http/src/System/Net/Http/Managed/HttpConnectionPool.cs

```diff
@@ -21,9 +21,16 @@ public HttpConnectionPool()
 
         public HttpConnection GetConnection()
         {
-            HttpConnection connection;
-            if (_idleConnections.TryTake(out connection))
+            while (_idleConnections.TryTake(out HttpConnection connection))
             {
+                if (connection.InitialReadCompleted)
+                {
+                    // We got a connection, but it was either already closed by the server or the
```

Comment:
I agree it would be nice.  It's also going to add some expense, as we'd need to create a continuation we otherwise wouldn't need to.  My expectation is that in cases where we care about performance/throughput, the majority of connections will get reused and very few will hit this (once the connection:close change goes in), in which case it should be relatively rare that connections we take from the pool have abruptly closed, and it's not worth paying extra for such cases.